internal/resources/app_endpoint_oidc.go

```diff
@@ -0,0 +1,387 @@
+	resp.TypeName = req.ProviderTypeName + "_app_endpoint_oidc_provider"
+}
+
+// Schema defines the Terraform schema for this resource.
+func (r *AppEndpointOidcProvider) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = AppEndpointOidcProviderSchema()
+}
+
+// ImportState imports a resource into Terraform state.
+func (r *AppEndpointOidcProvider) ImportState(
+	ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse,
+) {
+	// The import ID should be in the format: organizationId,projectId,clusterId,appServiceId,app_endpoint_name,provider_id
+	resource.ImportStatePassthroughID(ctx, path.Root("app_endpoint_name"), req, resp)
+}
+
+func (r *AppEndpointOidcProvider) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.Data = data
+}
+
+// Create creates a new OIDC provider and stores provider_id.
+func (r *AppEndpointOidcProvider) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.AppEndpointOidcProvider
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	organizationId := plan.OrganizationId.ValueString()
+	projectId := plan.ProjectId.ValueString()
+	clusterId := plan.ClusterId.ValueString()
+	appServiceId := plan.AppServiceId.ValueString()
+	appEndpointName := plan.AppEndpointName.ValueString()
+
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s/oidcProviders",
+		r.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+		appServiceId,
+		appEndpointName,
+	)
+
+	payload := buildAppEndpointOIDCProviderPayload(plan)
+
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPost, SuccessStatus: http.StatusCreated}
+	res, err := r.Client.ExecuteWithRetry(ctx, cfg, payload, r.Token, nil)
+	if err != nil {
+		resp.Diagnostics.AddError("Error Creating OIDC Provider", api.ParseError(err))
+		return
+	}
+	initOidcProviderNullsBeforeRefresh(&plan)
+
+	// Capture providerId from response
+	var created api.AppEndpointOIDCProviderResponse
+	var providerId string
+	if err := json.Unmarshal(res.Body, &created); err != nil {
+		resp.Diagnostics.AddError("Error unmarshalling create OIDC Provider response", api.ParseError(err))
+		return
+	}
+	if created.ProviderID != "" {
+		providerId = created.ProviderID
+		plan.ProviderId = types.StringValue(created.ProviderID)
```

Comment:
this is already done in `mapResponseToState` so no need for this check

Reply:
I've removed that check from mapResponseToState and kept it in the create as I think its mainly relevant for create

Reply:
https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/396/commits/23e6765c89e2ee3e9e3c07cb267733e05126f4c2